enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,995 @@
+
+##### Terraform variables and TFVarsSources
+
+The terraform `config` struct will need to be modified
+to include `VCenters`, `DeploymentZones` and `FailureDomains`
+as those structs contain the required parameters to
+provision a control plane node into a specific defined
+region and zone.
+
+```golang
+
+type config struct {
+	VSphereURL        string          `json:"vsphere_url"`
+  ...
+
+  VCenters []vtypes.VCenterSpec `json:"vcenters"`
+  DeploymentZones []vtypes.DeploymentZoneSpec `json:"deploymentZones"`
+  FailureDomains []vtypes.FailureDomainSpec `json:"failureDomains"`
+}
+```
+
+#### Machine and MachineSet
+
+The control plane
+[Machines](https://github.com/openshift/installer/blob/b0b96468893db2240e82ba2aa0935679c8c49201/pkg/asset/machines/vsphere/machines.go#L19-L64)
+will need to be modified to create a Machine per `DeploymentZone`.
+
+For each `DeploymentZone` an additional `MachineSet` for
+the compute nodes will need to be created.
+
+##### OVA import
+
+For each `DeploymentZone` a ova import will need to occur.
+If there is only a single zone then a single import will be required.
+
+### User Stories
+
+- https://issues.redhat.com/browse/RFE-845
+- https://issues.redhat.com/browse/OCPPLAN-4927
+- https://issues.redhat.com/browse/SPLAT-320
+
+### API Extensions
+
+
+### Risks and Mitigations
+
+- The out-of-tree CCM is required for this work. It will need to be enabled at
+installation time.
+
+## Design Details
+
+The vSphere platform spec and configuration will change to include `vcenters`, `DeploymentZones` and `FailureDomains`. The `MachinePool` will also add an
+optional `DeploymentZones` field.
+
+`FailureDomains` will contain a unique name including the following parameters:
+
+- region
+  - name
+  - type
+  - tagCategory
+- zone
+  - name
+  - type
+  - tagCategory
+- topology
+  - datacenter
+  - cluster
+  - networks
+  - datastore
+
+`DeploymentZones` will contain a unique name including the following parameters:
+
+- server
+- failureDomain
+- placement contstraints which includes:
+  - resource pool
+  - folder
+
+The vSphere platform spec will be the default configuration for the cloud-config
+adding the additional datacenters from the machinepool.
+
+The master virtual machines will be provisioned with terraform per datacenter
+and cluster. In the case of workers multiple machinesets will need to be
+configured - one per datacenter/cluster pair.
+
+### install-config
+
+```yaml
+apiVersion: v1
+baseDomain: example.com
+controlPlane:
+  name: master
+  platform:
+    vsphere:
+      deploymentZones:
+      - region-dc1-zone-c1
+      - region-dc1-zone-c2
+      - region-dc1-zone-c3
+      cpus: 8
+      coresPerSocket: 2
+      memoryMB: 24576
+      osDisk:
+        diskSizeGB: 512
+  replicas: 3
+compute:
+- name: worker
+  platform:
+    vsphere:
+      deploymentZones:
+      - region-dc1-zone-c1
+      - region-dc1-zone-c2
+      - region-dc1-zone-c3
```

Comment:
I believe these names are supposed to match the `name`s in `platform.vSphere.deploymentZones`.